### ISSUE TITLE
panic: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by therealdjryan** [containers/podman#11897]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

With some regularity the podman service panics and dies:
time="2021-10-08T10:47:51-04:00" level=debug msg="overlay: mount_data=,lowerdir=/tmp/containers-user-222/storage/overlay/l/LDZQDDWIGEXUFKJHMVHRCVC4UR:/tmp/containers-user-222/storage/overlay/l/FQVVABFJ5SX5W3XMIU7BSBGKSO:/tmp/containers-user-222/storage/overlay/l/IAMENZM2ADH6P6DRC43RGD5S6X:/tmp/containers-user-222/storage/overlay/l/57P37I4HFLDYDOT4UOKABU2M4D:/tmp/containers-user-222/storage/overlay/l/WWTIXKXIFCJQNNNTS577Y6WBCS:/tmp/containers-user-222/storage/overlay/l/IWXXO565WVUJI3UVVQLTYSOHTT:/tmp/containers-user-222/storage/overlay/l/I4RJZ7KPZE55SSZFFNCODUD473:/tmp/containers-user-222/storage/overlay/l/6GZVDLJCVRXSJHJLFDU47NITYO:/tmp/containers-user-222/storage/overlay/l/SRBAVIPGNJ4JNUVEMPJRW5UK5E:/tmp/containers-user-222/storage/overlay/l/IQWD7IIBU2GNOZEGG23V6JMLGL:/tmp/containers-user-222/storage/overlay/l/NL6VRI3LKWNGJ7VJKVQKX3L6UE:/tmp/containers-user-222/storage/overlay/l/JQJT7KEBNFRFUCDZDISJ72L3GG,upperdir=/tmp/containers-user-222/storage/overlay/d801ac2aae53f197f832c553ad2edf578d65753aa1b48508f416835a993d5a96/diff,workdir=/tmp/containers-user-222/storage/overlay/d801ac2aae53f197f832c553ad2edf578d65753aa1b48508f416835a993d5a96/work"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xa0 pc=0x557977aee776]
goroutine 5378438 [running]:
panic(0x557978bdc480, 0x5579799b8f20)
    /usr/lib/golang/src/runtime/panic.go:1064 +0x545 fp=0xc0005b5a30 sp=0xc0005b5968 pc=0x5579771b34a5
runtime.panicmem(...)
    /usr/lib/golang/src/runtime/panic.go:212
runtime.sigpanic()
    /usr/lib/golang/src/runtime/signal_unix.go:742 +0x413 fp=0xc0005b5a60 sp=0xc0005b5a30 pc=0x5579771ca513
github.com/containers/podman/vendor/github.com/containers/storage.(*store).canUseShifting(0xc00016d080, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x6)
    /builddir/build/BUILD/containers-podman-47dc7d7/_build/src/github.com/containers/podman/vendor/github.com/containers/storage/store.go:940 +0x36 fp=0xc0005b5a90 sp=0xc0005b5a60 pc=0x557977aee776
github.com/containers/podman/vendor/github.com/containers/storage.(*store).Mount(0xc00016d080, 0xc000574b00, 0x40, 0x0, 0x0, 0xc000bb2320, 0x0, 0x0, 0x5579771cb556)
    /builddir/build/BUILD/containers-podman-47dc7d7/_build/src/github.com/containers/podman/vendor/github.com/containers/storage/store.go:2711 +0x266 fp=0xc0005b5ba0 sp=0xc0005b5a90 pc=0x557977b00846
github.com/containers/podman/libpod.(*storageService).MountContainerImage(0xc000156570, 0xc0007b9e40, 0x40, 0x0, 0x0, 0x0, 0x0)
    /builddir/build/BUILD/containers-podman-47dc7d7/_build/src/github.com/containers/podman/libpod/storage.go:226 +0x235 fp=0xc0005b5c78 sp=0xc0005b5ba0 pc=0x557978280d15
github.com/containers/podman/libpod.(*Container).mount(0xc0007480a0, 0x81, 0x0, 0x0, 0x0)
    /builddir/build/BUILD/containers-podman-47dc7d7/_build/src/github.com/containers/podman/libpod/container_internal.go:2038 +0x70 fp=0xc0005b5d50 sp=0xc0005b5c78 pc=0x5579781f5910
github.com/containers/podman/libpod.(*Container).mountStorage(0xc0007480a0, 0x0, 0x0, 0xc0000dd8c0, 0xc0006f57a0)
    /builddir/build/BUILD/containers-podman-47dc7d7/_build/src/github.com/containers/podman/libpod/container_internal.go:1522 +0x627 fp=0xc0005b5f20 sp=0xc0005b5d50 pc=0x5579781ef827
github.com/containers/podman/libpod.(*Container).prepare.func2(0xc000546e10, 0xc0007480a0, 0xc000c1a640, 0xc000c1a630, 0xc000546d98)
    /builddir/build/BUILD/containers-podman-47dc7d7/_build/src/github.com/containers/podman/libpod/container_internal_linux.go:121 +0x75 fp=0xc0005b5fb8 sp=0xc0005b5f20 pc=0x5579782a1fd5
runtime.goexit()
    /usr/lib/golang/src/runtime/asm_amd64.s:1374 +0x1 fp=0xc0005b5fc0 sp=0xc0005b5fb8 pc=0x5579771eadc1
created by github.com/containers/podman/libpod.(*Container).prepare
    /builddir/build/BUILD/containers-podman-47dc7d7/_build/src/github.com/containers/podman/libpod/container_internal_linux.go:119 +0x1eb

goroutine 1 [chan receive, 3946 minutes]:
runtime.gopark(0x557978e0fcb0, 0xc0005c1df8, 0x170e, 0x2)
    /usr/lib/golang/src/runtime/proc.go:306 +0xe6 fp=0xc0008119c0 sp=0xc0008119a0 pc=0x5579771b6586
runtime.chanrecv(0xc0005c1da0, 0xc000673ab8, 0xc000000101, 0x557978564094)
    /usr/lib/golang/src/runtime/chan.go:577 +0x36f fp=0xc000811a50 sp=0xc0008119c0 pc=0x55797718346f
runtime.chanrecv1(0xc0005c1da0, 0xc000673ab8)
    /usr/lib/golang/src/runtime/chan.go:439 +0x2b fp=0xc000811a80 sp=0xc000811a50 pc=0x5579771830ab
github.com/containers/podman/pkg/api/server.(*APIServer).Serve(0xc000463e60, 0x0, 0xc00049dc38)
    /builddir/build/BUILD/containers-podman-47dc7d7/_build/src/github.com/containers/podman/pkg/api/server/server.go:227 +0x174 fp=0xc000811ad8 sp=0xc000811a80 pc=0x5579785640b4
github.com/containers/podman/cmd/podman/system.restService(0x7ffeefa90b21, 0x12, 0x0, 0x5579799e6080, 0xc00017df00, 0x557979a75260, 0x0, 0x0)
    /builddir/build/BUILD/containers-podman-47dc7d7/_build/src/github.com/containers/podman/cmd/podman/system/service_abi.go:85 +0x247 fp=0xc000811c78 sp=0xc000811ad8 pc=0x557978572967
github.com/containers/podman/cmd/podman/system.service(0x5579799e6080, 0xc0002163c0, 0x1, 0x3, 0x0, 0x0)
    /builddir/build/BUILD/containers-podman-47dc7d7/_build/src/github.com/containers/podman/cmd/podman/system/service.go:98 +0x1b1 fp=0xc000811d28 sp=0xc000811c78 pc=0x557978572031
github.com/containers/podman/vendor/github.com/spf13/cobra.(*Command).execute(0x5579799e6080, 0xc0000de040, 0x3, 0x3, 0x5579799e6080, 0xc0000de040)
    /builddir/build/BUILD/containers-podman-47dc7d7/_build/src/github.com/containers/podman/vendor/github.com/spf13/cobra/command.go:852 +0x47c fp=0xc000811e00 sp=0xc000811d28 pc=0x5579778260dc
github.com/containers/podman/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x5579799e9a00, 0xc0000d8020, 0x557978bd8ee0, 0x557979aa9928)
    /builddir/build/BUILD/containers-podman-47dc7d7/_build/src/github.com/containers/podman/vendor/github.com/spf13/cobra/command.go:960 +0x375 fp=0xc000811ed8 sp=0xc000811e00 pc=0x557977826c55
github.com/containers/podman/vendor/github.com/spf13/cobra.(*Command).Execute(...)
    /builddir/build/BUILD/containers-podman-47dc7d7/_build/src/github.com/containers/podman/vendor/github.com/spf13/cobra/command.go:897
github.com/containers/podman/vendor/github.com/spf13/cobra.(*Command).ExecuteContext(...)
    /builddir/build/BUILD/containers-podman-47dc7d7/_build/src/github.com/containers/podman/vendor/github.com/spf13/cobra/command.go:890
main.Execute()
    /builddir/build/BUILD/containers-podman-47dc7d7/_build/src/github.com/containers/podman/cmd/podman/root.go:90 +0xee fp=0xc000811f48 sp=0xc000811ed8 pc=0x557978587eee
main.main()
    /builddir/build/BUILD/containers-podman-47dc7d7/_build/src/github.com/containers/podman/cmd/podman/main.go:38 +0x94 fp=0xc000811f88 sp=0xc000811f48 pc=0x557978587774
runtime.main()
    /usr/lib/golang/src/runtime/proc.go:204 +0x209 fp=0xc000811fe0 sp=0xc000811f88 pc=0x5579771b6189
runtime.goexit()

goroutine 2 [force gc (idle), 3 minutes]:
runtime.gopark(0x557978e0fec8, 0x557979a70000, 0x1411, 0x1)
    /usr/lib/golang/src/runtime/proc.go:306 +0xe6 fp=0xc000074fb0 sp=0xc000074f90 pc=0x5579771b6586
runtime.goparkunlock(...)
    /usr/lib/golang/src/runtime/proc.go:312
runtime.forcegchelper()
    /usr/lib/golang/src/runtime/proc.go:255 +0xc5 fp=0xc000074fe0 sp=0xc000074fb0 pc=0x5579771b6425
runtime.goexit()
    /usr/lib/golang/src/runtime/asm_amd64.s:1374 +0x1 fp=0xc000074fe8 sp=0xc000074fe0 pc=0x5579771eadc1
created by runtime.init.7
    /usr/lib/golang/src/runtime/proc.go:243 +0x37

goroutine 3 [GC sweep wait]:
runtime.gopark(0x557978e0fec8, 0x557979a72800, 0x140c, 0x1)
    /usr/lib/golang/src/runtime/proc.go:306 +0xe6 fp=0xc0000757a8 sp=0xc000075788 pc=0x5579771b6586
runtime.goparkunlock(...)
    /usr/lib/golang/src/runtime/proc.go:312
runtime.bgsweep(0xc000054070)
    /usr/lib/golang/src/runtime/mgcsweep.go:182 +0x145 fp=0xc0000757d8 sp=0xc0000757a8 pc=0x5579771a19c5
runtime.goexit()
    /usr/lib/golang/src/runtime/asm_amd64.s:1374 +0x1 fp=0xc0000757e0 sp=0xc0000757d8 pc=0x5579771eadc1
created by runtime.gcenable
    /usr/lib/golang/src/runtime/mgc.go:217 +0x5e

goroutine 4 [sleep]:
runtime.gopark(0x557978e0fec8, 0x557979a73260, 0x1313, 0x2)
    /usr/lib/golang/src/runtime/proc.go:306 +0xe6 fp=0xc000075f20 sp=0xc000075f00 pc=0x5579771b6586
runtime.goparkunlock(...)
    /usr/lib/golang/src/runtime/proc.go:312
runtime.scavengeSleep(0x35ca84d, 0x21d793)
    /usr/lib/golang/src/runtime/mgcscavenge.go:241 +0xc5 fp=0xc000075f78 sp=0xc000075f20 pc=0x55797719f845
runtime.bgscavenge(0xc000054070)
    /usr/lib/golang/src/runtime/mgcscavenge.go:366 +0x1ed fp=0xc000075fd8 sp=0xc000075f78 pc=0x55797719fa6d
runtime.goexit()
    /usr/lib/golang/src/runtime/asm_amd64.s:1374 +0x1 fp=0xc000075fe0 sp=0xc000075fd8 pc=0x5579771eadc1
created by runtime.gcenable
    /usr/lib/golang/src/runtime/mgc.go:218 +0x85
**Steps to reproduce the issue:**

1.
The podman service is run like this:
/usr/bin/podman --log-level=debug system service -t 0 tcp:hostname:23750

2.
up to 8 containers are periodically rebuilt using podman-compose
3.
crash doesn't corespond with any particular activity
**Describe the results you received:**
panic

**Describe the results you expected:**
no panic

**Additional information you deem important (e.g. issue happens only occasionally):**
occasionally

**Output of `podman version`:**
```
Version:      3.2.3
API Version:  3.2.3
Go Version:   go1.15.7
Built:        Thu Jul 29 11:02:43 2021
OS/Arch:      linux/amd64
```

**Output of `podman info --debug`:**

```
host:
  arch: amd64
  buildahVersion: 1.21.3
  cgroupControllers: []
  cgroupManager: cgroupfs
  cgroupVersion: v1
  conmon:
    package: conmon-2.0.29-1.module+el8.4.0+11822+6cc1e7d7.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.29, commit: ae467a0c8001179d4d0adf4ada381108a893d7ec'
  cpus: 8
  distribution:
    distribution: '"rhel"'
    version: "8.4"
  eventLogger: file
  hostname: gpuigrid.broadinstitute.org
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 66
      size: 1
    - container_id: 1
      host_id: 22200000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 222
      size: 1
    - container_id: 1
      host_id: 22200000
      size: 65536
  kernel: 4.18.0-305.19.1.el8_4.x86_64
  linkmode: dynamic
  memFree: 314249216
  memTotal: 12345868288
  ociRuntime:
    name: runc
    package: runc-1.0.0-74.rc95.module+el8.4.0+11822+6cc1e7d7.x86_64
    path: /usr/bin/runc
    version: |-
      runc version spec: 1.0.2-dev
      go: go1.15.13
      libseccomp: 2.5.1
  os: linux
  remoteSocket:
    exists: true
    path: /run/user/222/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_NET_RAW,CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.1.8-1.module+el8.4.0+11822+6cc1e7d7.x86_64
    version: |-
      slirp4netns version 1.1.8
      commit: d361001f495417b880f20329121e3aa431a8f90f
      libslirp: 4.3.1
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.1
  swapFree: 5813620736
  swapTotal: 6350172160
  uptime: 265h 39m 6.09s (Approximately 11.04 days)
registries:
  search:
  - registry.access.redhat.com
  - registry.redhat.io
  - docker.io
store:
  configFile: /home/unix/releng/.config/containers/storage.conf
  containerStore:
    number: 14
    paused: 0
    running: 8
    stopped: 6
  graphDriverName: overlay
  graphOptions:
    overlay.mount_program:
      Executable: /usr/bin/fuse-overlayfs
      Package: fuse-overlayfs-1.6-1.module+el8.4.0+11822+6cc1e7d7.x86_64
      Version: |-
        fusermount3 version: 3.2.1
        fuse-overlayfs: version 1.6
        FUSE library version 3.2.1
        using FUSE kernel interface version 7.26
  graphRoot: /tmp/containers-user-222/storage
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 20
  runRoot: /run/user/222/containers
  volumePath: /tmp/containers-user-222/storage/volumes
version:
  APIVersion: 3.2.3
  Built: 1627570963
  BuiltTime: Thu Jul 29 11:02:43 2021
  GitCommit: ""
  GoVersion: go1.15.7
  OsArch: linux/amd64
  Version: 3.2.3

```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
podman-3.2.3-0.11.module+el8.4.0+12050+ef972f71.x86_64
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/master/troubleshooting.md)**

No

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by nalind** [comment]
This looks a lot like https://github.com/containers/podman/issues/10535.

**Comment by therealdjryan** [comment]
@mheon @nalind, If there's more I can do to help find the cause of this let me know.

**Comment by nalind** [comment]
The containers/storage bug was fixed in its v1.32.2 release.  That version, and later versions, are used in podman's 3.3 and 3.4 branches.  I believe one of those versions will be included in the "container-tools" module in RHEL 8.5.

**Comment by vrothberg** [comment]
Closing as it's fixed in the main branch.
